src/EventMachine.ts

```diff
@@ -1,10 +1,38 @@
 import { Store } from "./store"
 import { Event } from "./types"
 
-export function getAnEvent(store: Store, allEvents: Event[]): Event | null {
-  const priorizedEvents = allEvents
-    .map((event) => ({ ...event, priority: event.probability(store) * Math.random() }))
-    .filter((event) => event.priority)
-    .sort((a, b) => b.priority - a.priority)
-  return  priorizedEvents.length ? priorizedEvents[0] : null
+export default function(store: Store, allEvents: Event[]) {
+  let timer: number
+
+  function getPriorizedEvents(): Event[] {
+    return allEvents
+      .map((event) => ({ ...event, priority: event.probability(store) * Math.random() }))
```

Comment:
Auch gerne erstmal so rein. Nur zum gemeinsamen Nachdenken: Auch die sicheren "GameOver" Events werden so nicht sofort passieren, oder? Bei einer "probility" von 1 würde ich erwarten, dass der Event sofort ausgelöst wird. Ich denke, "Math.random()" muss eher in "filter()". Ich bastel gern mal was, wenn Du das eingecheckt hast.

Reply:
Stimmt, eine probability von 1 kommt nicht unmittelbar dran, sondern erst wenn das nächste Event gesucht wird - und das passiert innerhalb von 1 Minute. So lange könnte man also z.B. ohne Budget weiter wirtschaften.